### PR TITLE
Add checks for CMake and Curl

### PR DESCRIPTION
**Pull request by jacobleeharris**
Adds checks for CMake and Curl which are needed by ps2sdk-ports to build.
It is best to ensure that these are installed before failing in the middle of the build.

### CODE REVIEW

**Comment by jacobleeharris** [comment]
Seems like CI failed on Unix-based builds due to some unrelated issues. This shouldn't really have affected what the CI is doing.

**Comment by uyjulian** [comment]
The scripts need executable permissions.

**Comment by PhoenixVX** [comment]
Fixed.

**Comment by uyjulian** [comment]
```
./build-all.sh: line 17: ../depends/check-cmake.sh: Permission denied
../depends/check-cmake.sh: Failed.
Error: Process completed with exit code 1.
```

To resolve this error, you need to set executable flags on the script.
This can be usually done with
```
chmod +x /path/to/file.sh
```

When you do `ls -l` on the file, it will show `x` flag:
```
$ ls -l /Volumes/public_share/ps2src/ps2dev/ps2dev/depends/check-gcc.sh 
-rwxr-xr-x 1 uyjulian uyjulian 164 Jun  4  2020 /Volumes/public_share/ps2src/ps2dev/ps2dev/depends/check-gcc.sh
```

**Comment by jacobleeharris** [comment]
Sorry about that, seems like using git's update-index to chmod the permissions isn't all that's needed to properly set the executable permissions. It should be good now.

File: README.md [hunk 1]
```diff
@@ -20,6 +20,18 @@ These scripts download (`git clone`) and install:
 ## Requirements
 
 1.  Install gcc/clang, make, cmake, patch, git, texinfo, flex, bison, gettext, wget, gsl, gmp, zlib, mpfr and mpc if you don't have those packages.
+-  Ubuntu
+```bash
+sudo apt -y install gcc make cmake patch git texinfo flex bison gettext libgsl-dev libgmp3-dev libmpfr-dev libmpc-dev zlib1g-dev
```
Comment (Florin9doi) [inline README.md:25]:
```suggestion
sudo apt -y install gcc g++ make cmake patch git texinfo flex bison gettext libgsl-dev libgmp3-dev libmpfr-dev libmpc-dev zlib1g-dev
```

File: README.md [hunk 2]
```diff
@@ -20,6 +20,18 @@ These scripts download (`git clone`) and install:
 ## Requirements
 
 1.  Install gcc/clang, make, cmake, patch, git, texinfo, flex, bison, gettext, wget, gsl, gmp, zlib, mpfr and mpc if you don't have those packages.
+-  Ubuntu
```
Comment (Florin9doi) [inline README.md:23]:
```suggestion
  - Ubuntu
```

File: README.md [hunk 3]
```diff
@@ -20,6 +20,18 @@ These scripts download (`git clone`) and install:
...
+```bash
+sudo apt -y install gcc make cmake patch git texinfo flex bison gettext libgsl-dev libgmp3-dev libmpfr-dev libmpc-dev zlib1g-dev
+```
+-  Fedora/CentOS
```
Comment (Florin9doi) [inline README.md:27]:
```suggestion
  - Fedora/CentOS
```

File: README.md [hunk 4]
```diff
@@ -20,6 +20,18 @@ These scripts download (`git clone`) and install:
...
+```bash
+sudo dnf -y install gcc make cmake patch git texinfo flex bison gettext gmp-devel mpfr-devel libmpc zlib-devel
+```
+-  Alpine
```
Comment (Florin9doi) [inline README.md:31]:
```suggestion
  - Alpine
```

**Comment by Florin9doi** [comment]
Fix : https://github.com/jacobleeharris/ps2dev/pull/1

**Comment by jacobleeharris** [comment]
Seems like even with your edits, Codacy checkstyle is still failing.

**Comment by uyjulian** [comment]
markdownlint was recently added, so you could use it locally to fix most of the issues

**Comment by uyjulian** [comment]
Closed in favor of #80

**Pull request closed without merging**